docs/vc_patch.md

```diff
@@ -0,0 +1,113 @@
+
+```bash
+tools/make_patch labels.sym constants.sym patched.gbc original.gbc vc.patch.template vc.patch
+```
+
+For example, this is what `make crystalvc` uses:
+
+```bash
+tools/make_patch pokecrystal11_vc.sym vc/pokecrystal11_vc.constants.sym pokecrystal11_vc.gbc pokecrystal11.gbc vc/pokecrystal11_vc.patch.template pokecrystal11_vc.patch
+```
+
+
+## Patch template syntax
+
+**Comments** start at a semicolon "`;`" and continue until the end of the line. They are output as-is, without interpreting commands.
+
+**Patch names** are contained in "`[`" brackets "`]`". They are output as-is, without interpreting commands.
+
+Patch names also set the **current patch label**. This is the label starting with "`.VC_`" followed by the patch name, with any invalid characters (not letters "`A-Z`", digits "`0-9`", underscore "`_`", at sign "`@`", or hash "`#`") converted to underscores "`_`". For example, the patch name "`[fight begin]`" corresponds to the patch label "`.VC_fight_begin`".
+
+**Commands** are contained in "`{`" braces "`}`". They are not output themselves, but may produce their own output when interpreted.
+
+Commands are interpreted with a series of arguments, separated by whitespace (spaces, tabs, or newlines). Leading and trailing whitespace is ignored; for example, "`{  hex  @  4  }`" is interpreted the same as "`{hex @ 4}`".
+
+**Arguments** evaluate to numeric values. They may be any of the following:
+
+- Literal numbers in decimal (base 10, e.g. "`42`"), hexadecimal (base 16, e.g. "`0x2a`"), or octal (base 8, e.g. "`052`"). They may start with a plus sign "`+`". Numbers may not be negative.
+- Comparison operators: "`==`" is 0, "`>`" is 1, "`<`" is 2, "`>=`" is 3, "`<=`" is 4, "`!=`" is 5, and "`||`" is 0x11.
+- Symbol names from the two `.sym` files provided to `make_patch` may evaluate as their bank-relative address, or their absolute offset in the ROM, depending on the command. They may also be followed by a plus sign and a literal number that gets added to the value.
+- "`@`" evaluates as the address or absolute offset of the current patch label.
+
+Any other characters are output as-is.
+
+
+## Patch template commands
+
+### `patch`, `PATCH`, `patch_`, or `PATCH_`
+
+Seeks the patched ROM contents between the current patch label, and the label which is the current patch label plus "`_End`". Outputs the bytes between those labels as a hexadecimal number "<code>0x<i>V</i></code>" for only one byte, or as a *value series* "<code>a<i>N</i>: <i>V1</i> <i>V2</i> [...] <i>VN</i></code>" for multiple bytes.
+
+An optional argument is an *offset* to add to the current patch label before gathering the contents between it and the end label.
+
+If the command name is all lowercase, the byte values use lowercase for hexadecimal digits A-F; if it is all uppercase, they use uppercase. If the command name ends in an underscore, a space is output after the colon preceding the values; if not, then it is not.
```

Comment:
This (or a different section, maybe the "concepts" section mentioned above) should explain why the different command names exist.

(opinion)
I would also just explain 1 format for each command (i.e. only mention and explain the lowercase one, make sure to include an example), while listing the rest along with the explanation of why they exist.